crates/oxide/src/scanner/auto_source_detection.rs

```diff
@@ -0,0 +1,54 @@
+use ignore::gitignore::{Gitignore, GitignoreBuilder};
+use std::sync;
+
+/// All the default rules for auto source detection.
+///
+/// This includes:
+///
+/// - Ignoring common content directories like `.git` and `node_modules`
+/// - Ignoring file extensions we definitely don't want to include like `.css` and `.scss`
+/// - Ignoring common binary file extensions like `.png` and `.jpg`
+/// - Ignoring common files like `yarn.lock` and `package-lock.json`
+///
+pub static RULES: sync::LazyLock<Gitignore> = sync::LazyLock::new(|| {
+    let mut builder = GitignoreBuilder::new("");
+
+    builder
+        .add_line(None, &format!("{{{}}}", IGNORED_CONTENT_DIRS.join(",")))
```

Comment:
Should this have a `/` at the end of the pattern?

Reply:
Ah, yep I think so. We learned that the inclusion of a slash in the gitignore rule makes it relative to the location of `.gitignore` but a stray `/` at the end does not seem to do that, it just makes it clear that it's a dir 👍 

<img width="677" alt="Screenshot 2025-03-25 at 11 56 55" src="https://github.com/user-attachments/assets/155b21a7-fcea-4383-bf67-f5068af73044" />

Reply:
Yeah, the trailing `/` forces a directory match. So it doesn't match a _file_ named `node_modules` for instance but does match a directory named `node_modules`